src/credentials/providers/ssoCredentialProvider.ts

```diff
@@ -23,27 +23,26 @@ export class SsoCredentialProvider {
     public async refreshCredentials(): Promise<Credentials> {
         try {
             const accessToken = await this.ssoAccessTokenProvider.accessToken()
-            const roleCredentials = await this.ssoClient
-                .getRoleCredentials({
-                    accountId: this.ssoAccount,
-                    roleName: this.ssoRole,
-                    accessToken: accessToken.accessToken,
-                })
+            const roleCredentials = await this.ssoClient.getRoleCredentials({
+                accountId: this.ssoAccount,
+                roleName: this.ssoRole,
+                accessToken: accessToken.accessToken,
+            })
 
             return {
                 accessKeyId: roleCredentials.roleCredentials!.accessKeyId!,
                 secretAccessKey: roleCredentials.roleCredentials!.secretAccessKey!,
                 sessionToken: roleCredentials.roleCredentials?.sessionToken,
             }
         } catch (err) {
-            if (err.code === 'UnauthorizedException') {
+            if ((err as { code: string }).code === 'UnauthorizedException') {
                 this.ssoAccessTokenProvider.invalidate()
             }
             vscode.window.showErrorMessage(
                 localize('AWS.message.credentials.sso.error', 'Failed to load SSO credentials. Try logging in again.')
             )
-            getLogger().error(err)
-            throw err
+            getLogger().error(err as Error)
+            throw err // TODO: remove throw? we are already handling the error by showing the user a message...
```

Comment:
typically the (assumed) reason is that it's used for control flow, sometimes by vscode itself.

Reply:
Yeah it's for control flow. Since the top-level credential module also shows an error notification we end up showing two different error messages. I'll leave it be since the notification is helpful, though showing two notifications isn't ideal.